src/javascripts/components/search.js

```diff
@@ -126,20 +142,44 @@ Search.prototype.init = function () {
     confirmOnBlur: false,
     autoselect: true,
     source: this.handleSearchQuery.bind(this),
-    onConfirm: this.handleOnConfirm,
+    onConfirm: this.handleOnConfirm.bind(this),
     templates: {
-      inputValue: this.inputValueTemplate,
-      suggestion: this.resultTemplate
+      inputValue: this.inputValueTemplate.bind(this),
+      suggestion: this.resultTemplate.bind(this)
     },
     tNoResults: function () { return statusMessage }
   })
 
-  // Ensure the Button (which is a background image of the wrapper) focuses the input when clicked.
   var $wrapper = $module.querySelector('.app-site-search__wrapper')
+  var $input = $module.querySelector('.app-site-search__input')
+
+  // Since the autocomplete does not have events we add additional listeners
+  // which allows for tracking of certain events.
+
+  // We want to wait a bit before firing events to indicate that
+  // someone is looking at a result and not that it's come up in passing.
+  var timeToWait = function () {
+    // We want to be able to reduce this timeout in order to make sure
+    // our tests do not run very slowly.
+    var timeout = window.__SITE_SEARCH_TRACKING_TIMEOUT
+    return (typeof timeout !== 'undefined') ? timeout : 2000 // milliseconds
+  }
+  var eventTimer
+  $input.addEventListener('input', function (event) {
+    clearTimeout(eventTimer)
+    eventTimer = setTimeout(function () {
+      trackInput($module, $input)
+    }, timeToWait())
+  })
+  $input.addEventListener('blur', function (event) {
```

Comment:
are we mising `import 'govuk-frontend/vendor/polyfills/Event' `

Reply:
Think we're OK in this instance since that is available in IE9+ https://caniuse.com/#search=event

Reply:
👍